### DISCUSSION TITLE
directory-maven-plugin:1.0 fails

### DISCUSSION THREAD

**Discussion by heatherita** [flowlogix/flowlogix#306] [Q&A] [answered]
When I try to compile using your flowlogix-bom in my parent pom.xml like this:

```
<parent>               
        <groupId>com.flowlogix</groupId>
        <artifactId>flowlogix-bom</artifactId>
        <version>4.0.7</version>
    </parent>
```

It fails with this message:

```
--- directory-maven-plugin:1.0:directory-of (directories) @ baw-website ---
------------------------------------------------------------------------
BUILD FAILURE
------------------------------------------------------------------------
Total time:  5.132 s
Finished at: 2022-10-06T09:50:16-04:00
------------------------------------------------------------------------
Failed to execute goal org.commonjava.maven.plugins:directory-maven-plugin:1.0:directory-of (directories) on project baw-website: Cannot find directory for project: com.flowlogix:flowlogix -> [Help 1]
```
I'll try to skip directory-maven-plugin:1.0 for now.

**Comment by lprimak** [comment, marked as answer]
Hi,

you are not supposed to use bom as a parent. 
See https://howtodoinjava.com/maven/maven-bom-bill-of-materials-dependency/ (quick google search yielded this)
or in https://github.com/lprimak/apps/blob/main/emailmanager/pom.xml has an example
the `apps` application also has skinny war example

**Comment by heatherita** [comment]
I am now using this syntax in many of my pom.xml:
```
 <dependencies>
    <dependency>
            <groupId>com.flowlogix</groupId>
            <artifactId>flowlogix-jee</artifactId>
            <version>${flowlogix.version}</version>
        </dependency>
    </dependencies>
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.flowlogix</groupId>
                <artifactId>flowlogix-bom</artifactId>
                <version>4.0.7</version>            
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

 I believe it's correct based on  https://howtodoinjava.com/maven/maven-bom-bill-of-materials-dependency/

**Comment by lprimak** [comment]
You could use `${flowlogix.version}` instead of `4.0.7`
FYI 5.0 will be released soon so I wouldn't even bother to update to 4.x and just go straight to 5.
Milestones are already available so use the latest one.

**Comment by heatherita** [comment]
I will upgrade but 5.0 does not come up as a choice in Netbeans when I set the repo to http://raw.github.com/flowlogix/flowlogix-m2/master. 

I plan to use ${flowlogix.version} but I'm hard-coding for testing.

Also,  I see that flowlogix-datamodel is at 4.0.6, but I assume that will be upgraded to 5.0 soon too.

thanks!

**Reply by lprimak** [reply]
http://raw.github.com/flowlogix/flowlogix-m2/master has been removed and no longer needed. use https://search.maven.org as NetBeans doesn't necessarily have the latest components downloaded

**Reply by heatherita** [reply]
This is incorrect. The correct url is `https://repo1.maven.org/maven2/`

Run `curl get https://search.maven.org/` to see what I mean
